### ISSUE TITLE
Codespaces does not work on Apple Silicon builds

### ISSUE THREAD

**Issue by ItalyPaleAle** [microsoft/vscode#111345]
Version: 1.52.0-exploration
Commit: f9b242d7e68aec36dc08ee7840d8f3a2b22b51d5
Date: 2020-11-25T18:56:00.184Z
Electron: 11.0.3
Chrome: 87.0.4280.67
Node.js: 12.18.3
V8: 8.7.220.25-electron.0
OS: Darwin arm64 20.1.0

Using Codespaces in VS Code Exploration for Apple Silicon, try to connect to a Codespace running on GitHub (already created using the browser)

This is the content of the **Log (Window)** output:

```
[2020-11-25 21:19:20.793] [renderer1] [error] [remote-connection] An error occurred in the very first connect attempt, it will be treated as a permanent error! Error:
[2020-11-25 21:19:20.793] [renderer1] [error] Error: Failed to start VS Code Remote server.
    at handleResponse (/Users/alessandro/.vscode-exploration/extensions/node_modules/vscode-jsonrpc/lib/main.js:436:1)
    at processMessageQueue (/Users/alessandro/.vscode-exploration/extensions/node_modules/vscode-jsonrpc/lib/main.js:263:1)
    at Immediate.<anonymous> (/Users/alessandro/.vscode-exploration/extensions/node_modules/vscode-jsonrpc/lib/main.js:247:1)
    at processImmediate (internal/timers.js:456:21)
[2020-11-25 21:19:20.805] [renderer1] [warning] Ignoring the error while validating workspace folder vscode-remote://vsonline%2B189d85ea-7a1f-49ef-aef7-52b80d10c796%2Baccountprovider%3Dgithub/home/vscode/workspace/prvt - Failed to start VS Code Remote server.
[2020-11-25 21:19:20.806] [renderer1] [error] Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/settings.json' (Error: Failed to start VS Code Remote server.): Error: Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/settings.json' (Error: Failed to start VS Code Remote server.)
    at _.doReadAsFileStream (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1798:809)
    at async _.readFile (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1797:933)
    at async file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:162
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at async m.loadConfiguration (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:322)
[2020-11-25 21:19:20.806] [renderer1] [error] Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/tasks.json' (Error: Failed to start VS Code Remote server.): Error: Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/tasks.json' (Error: Failed to start VS Code Remote server.)
    at _.doReadAsFileStream (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1798:809)
    at async _.readFile (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1797:933)
    at async file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:162
    at async Promise.all (index 0)
    at async Promise.all (index 1)
    at async m.loadConfiguration (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:322)
[2020-11-25 21:19:20.806] [renderer1] [error] Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/launch.json' (Error: Failed to start VS Code Remote server.): Error: Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/launch.json' (Error: Failed to start VS Code Remote server.)
    at _.doReadAsFileStream (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1798:809)
    at async _.readFile (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1797:933)
    at async file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:162
    at async Promise.all (index 1)
    at async Promise.all (index 1)
    at async m.loadConfiguration (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:322)
[2020-11-25 21:19:20.806] [renderer1] [error] Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/settings.json' (Error: Failed to start VS Code Remote server.): Error: Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/settings.json' (Error: Failed to start VS Code Remote server.)
    at _.doReadAsFileStream (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1798:809)
    at async _.readFile (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1797:933)
    at async file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:162
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at async m.loadConfiguration (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:322)
    at async Promise.all (index 0)
[2020-11-25 21:19:20.806] [renderer1] [error] Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/tasks.json' (Error: Failed to start VS Code Remote server.): Error: Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/tasks.json' (Error: Failed to start VS Code Remote server.)
    at _.doReadAsFileStream (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1798:809)
    at async _.readFile (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1797:933)
    at async file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:162
    at async Promise.all (index 0)
    at async Promise.all (index 1)
    at async m.loadConfiguration (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:322)
    at async Promise.all (index 0)
[2020-11-25 21:19:20.806] [renderer1] [error] Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/launch.json' (Error: Failed to start VS Code Remote server.): Error: Unable to read file 'vscode-remote://vsonline+189d85ea-7a1f-49ef-aef7-52b80d10c796+accountprovider=github/home/vscode/workspace/prvt/.vscode/launch.json' (Error: Failed to start VS Code Remote server.)
    at _.doReadAsFileStream (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1798:809)
    at async _.readFile (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1797:933)
    at async file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:162
    at async Promise.all (index 1)
    at async Promise.all (index 1)
    at async m.loadConfiguration (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:3706:322)
    at async Promise.all (index 0)
[2020-11-25 21:19:20.819] [renderer1] [error] Failed to start VS Code Remote server.: Error: Failed to start VS Code Remote server.
    at handleResponse (/Users/alessandro/.vscode-exploration/extensions/node_modules/vscode-jsonrpc/lib/main.js:436:1)
    at processMessageQueue (/Users/alessandro/.vscode-exploration/extensions/node_modules/vscode-jsonrpc/lib/main.js:263:1)
    at Immediate.<anonymous> (/Users/alessandro/.vscode-exploration/extensions/node_modules/vscode-jsonrpc/lib/main.js:247:1)
    at processImmediate (internal/timers.js:456:21)
[2020-11-25 21:19:22.113] [renderer1] [error] spawn git ENOENT: Error: spawn git ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)

```

And this is the log of the **VS Codespaces** extension:

```
[Client V] Initializing authentication
[Client I] Trace log: /var/folders/yv/kj13y94x0jl5hbwqm8m1n4qh0000gn/T/vsonline_logs/20201126_051914_16063679540190_VSCode.log
[GitHubAuthentication V] Attempting to load session - login = false
[GitHubAuthentication V] Attempting to load session - login = false
[GitHubAuthentication V] Successfully loaded session
[GitHubAuthentication V] Successfully loaded session
[Client I] Extension activating {"isInternal":false,"isRunningInCloudEnvironment":false,"serviceEndpoint":"https://online.visualstudio.com/api/v1/environments"}
[EnvironmentDetailsExplorer I] Updating environment details
[Client I] Monitoring git changes
[VSOnlineResolver I] Resolving: 189d85ea-7a1f-49ef-aef7-52b80d10c796 with params accountProvider=GitHub
[VSOnlineResolver I] Fetching codespace details
[VSOnlineResolver I] Successfully fetched codespace details
[EnvironmentDetailsExplorer I] Updating environment details
[VSOnlineResolver I] Attempting to connect to codespace
[VSOnlineResolver I] Connected to remote host...
[EnvironmentDetailsExplorer I] Updating environment details
[WorkspaceApi V] get workspace info -> https://prod.liveshare.vsengsaas.visualstudio.com/api/v1.2/workspace/FE221F840C2B768CA957B9820DE027A93CA5
[FetchRequest:1 V] GET - fetch start
[FetchRequest:1 V] GET - fetch response. (url=https://prod.liveshare.vsengsaas.visualstudio.com/api/v1.2/workspace/FE221F840C2B768CA957B9820DE027A93CA5;status=200)
[WorkspaceApi V] get workspace access -> https://prod.liveshare.vsengsaas.visualstudio.com/api/v1.2/workspace/111C0E36B9D55F1246D4129322D4801285FC/user
[FetchRequest:2 V] PUT - fetch start
[FetchRequest:2 V] PUT - fetch response. (url=https://prod.liveshare.vsengsaas.visualstudio.com/api/v1.2/workspace/111C0E36B9D55F1246D4129322D4801285FC/user;status=200)
[WorkspaceClient V] connecting to relay link:sb://vsls-prod-ins-usw2-private-relay.servicebus.windows.net/111c0e36b9d55f1246d4129322d4801285fc--9fc95651-548c-40dc-84bf-04c3120c1603
[WorkspaceClient V] creating node SSH...
[WorkspaceClient V] authenticating...
[WorkspaceClient V] joining...
[WorkspaceClient V] < version.exchangeVersions()
[WorkspaceClient V] > version.exchangeVersions() succeeded t:55.0
[WorkspaceClient I] Host version: {"version":"1.0.3231.55444","protocolVersion":"2.2","platformName":"Linux 5.4.0","platformVersion":"5.4.0.1031"}
[WorkspaceClient V] < workspace.joinWorkspace()
[WorkspaceClient V] > workspace.servicesChanged: 
[WorkspaceClient V] > workspace.progressUpdated: 
[WorkspaceClient V] > workspace.joinWorkspace() succeeded t:60.0
[WorkspaceClient V] join completed -> time:184.0 session:12 conversation:9fc95651-548c-40dc-84bf-04c3120c1603
[WorkspaceClient V] < workspace.registerServices()
[WorkspaceClient V] > workspace.registerServices() succeeded t:54.0
[VSOnlineResolver I] Shared git credential service
[VSOnlineResolver I] Configuring Codespace...
[WorkspaceServices V] < IEnvironmentConfigurationService.configureEnvironment()
[WorkspaceServices V] > IEnvironmentConfigurationService.configureEnvironment() succeeded t:57.0
[VSOnlineResolver I] Forwarding existing ports...
[WorkspaceServices V] < serverSharing.getSharedServers()
[WorkspaceServices V] > serverSharing.getSharedServers() succeeded t:53.0
[VSOnlineResolver I] Starting remote server...
[WorkspaceServices V] < IVSCodeServerHostService.getCodespaceStateFilePath()
[WorkspaceServices V] < IVSCodeServerHostService.startRemoteServer()
[WorkspaceServices V] > IVSCodeServerHostService.getCodespaceStateFilePath() succeeded t:57.0
[WorkspaceServices V] < file.readText()
[WorkspaceServices V] > file.readText() succeeded t:63.0
[WorkspaceServices V] < IVSCodeServerHostService.getDevcontainerStateFilePath()
[WorkspaceServices V] > IVSCodeServerHostService.getDevcontainerStateFilePath() succeeded t:50.0
[WorkspaceServices V] < file.readText()
[WorkspaceServices V] > file.readText() succeeded t:49.0
[WorkspaceServices E] > IVSCodeServerHostService.startRemoteServer() error: undefined
[VSOnlineResolver E] Resolve failed: Error: Failed to start VS Code Remote server.
[Client I] Error (FileSystemError): Failed to start VS Code Remote server.
[WorkspaceClient V] > workspace.servicesChanged: 

```

Does this issue occur when all extensions are disabled?: N/A

**Comment by deepak1556** [comment]
This should be fixed in latest release a31f124319, can you confirm. Thanks!

**Comment by ItalyPaleAle** [comment]
Confirming Codespaces works on the latest build. Thanks!
